### PR TITLE
feat: Adds tooltip to reconnect datasources list.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
Adds tooltip to the icon next to data source name

This is helpful when data source name is long and truncated.

Fixes # (issue) [16067](https://github.com/appsmithorg/appsmith/issues/16067)

Media
Screenshot 2023-04-04 at 3 50 11 PM

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Added cypress test and confirmed locally.
Steps to reproduce are in the issue link above

- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/components/DatasourceListItem.tsx [hunk 1]
```diff
@@ -69,11 +70,14 @@ function ListItemWrapper(props: {
           >
             {ds.name}
           </Text>
-          <Icon
-            fillColor={ds.isConfigured ? Colors.GREEN : Colors.ERROR_RED}
-            name={ds.isConfigured ? "oval-check" : "info"}
-            size={IconSize.MEDIUM}
-          />
+          <Tooltip2 content={ds.name} placement="right">
```
Comment (akash-codemonk) [inline app/client/src/pages/Editor/gitSync/components/DatasourceListItem.tsx:73]: Could we use the design system tooltip for this?

Comment (rahulbarwal) [reply app/client/src/pages/Editor/gitSync/components/DatasourceListItem.tsx:73]: Done.

**Review by akash-codemonk — APPROVED** [review]

**Comment by rahulbarwal** [comment]
@ramsaptami Can you please do QA fro this PR?

**Comment by Richarex** [comment]
Tested this on the DP url and it looks good to me. The tooltips are displayed with the ```i``` icon for reconnect datasources list SS:-

**Review by akash-codemonk — APPROVED** [review]

**Pull request merged**